### PR TITLE
Graph widget

### PR DESCRIPTION
**Pull request by druskus20**
## Description

This PR implements circular progress widgets as mentioned in #329 

## Usage

```lisp
(box :class "bar"
       :orientation "v"
       :space-evenly true
    (graph :class "graph"
           :height 200
           :width 200
           :value cpuvar 
           :max 20
           :min 0
           :dynamic true
           :thickness 2
           :line-style "miter"
           :time-range "10s"))) 
```

### Showcase

out
out
out

## Checklist

Please make sure you can check all the boxes that apply to this PR.

- [x] All widgets I've added are correctly documented.
- [x] The documentation in the `docs/content/main` directory has been adjusted to reflect my changes.
- [x] I used `cargo fmt` to automatically format all code before committing

### CODE REVIEW

File: crates/eww/src/widgets/graph.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+impl ObjectSubclass for GraphPriv {
+    type ParentType = gtk::Bin;
+    type Type = Graph;
+
+    const NAME: &'static str = "Graph";
+}
+
+impl Graph {
+    pub fn new() -> Self {
+        glib::Object::new::<Self>(&[]).expect("Failed to create MyAwesome Widget")
```
Comment (legendofmiracles) [inline crates/eww/src/widgets/graph.rs:97]:
```suggestion
        glib::Object::new::<Self>(&[]).expect("Failed to create graph Widget")
```
My awesome? Based on context this should be Graph, right?

File: crates/eww/src/widgets/graph.rs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                obj.queue_draw();
+            }
+            "thickness" => {
+                self.thickness.replace(value.get().unwrap());
+            }
+            "range" => {
+                self.range.replace(value.get().unwrap());
+            }
+
+            x => panic!("Tried to set inexistant property of G()raph: {}", x,),
```
Comment (legendofmiracles) [inline crates/eww/src/widgets/graph.rs:73]:
```suggestion
            x => panic!("Tried to set inexistant property of Graph: {}", x,),
```

**Comment by druskus20** [comment]
The PR should be working as of right now, however, there's a couple of things worth adding:
- [ ] Dynamic Y range
- [ ] Allow specifying if the graph should go downwards or upwards

**Comment by elkowar** [comment]
Very epic!

**Pull request merged**